### PR TITLE
feat(elbv2): add name validation for target group and load balancer names

### PR DESCRIPTION
**Pull request by mina-asham**
- Target group naming rules: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-targetgroup.html#cfn-elasticloadbalancingv2-targetgroup-name
- Load balancer naming rules: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-loadbalancer.html#cfn-elasticloadbalancingv2-loadbalancer-name

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-elasticloadbalancingv2/test/nlb/load-balancer.test.ts [hunk 1]
```diff
@@ -241,6 +241,76 @@ describe('tests', () => {
...
+    const stack = new cdk.Stack();
+    const vpc = new ec2.Vpc(stack, 'Stack');
+
+    // WHEN/THEN
+    expect(() =>
+      new elbv2.NetworkLoadBalancer(stack, 'NLB', {
+        loadBalancerName: 'internal-myLoadBalancer',
+        vpc,
+      }),
+    ).toThrow();
```
Comment (corymhall) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/test/nlb/load-balancer.test.ts:255]: Can you add the expected error message to the assertions in all these?

```ts
toThrow(/some error message/);
```

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts [hunk 2]
```diff
@@ -245,6 +245,10 @@ export abstract class TargetGroupBase extends CoreConstruct implements ITargetGr
       this.setAttribute('deregistration_delay.timeout_seconds', baseProps.deregistrationDelay.toSeconds().toString());
     }
 
+    if (!cdk.Token.isUnresolved(baseProps.targetGroupName) && baseProps.targetGroupName !== undefined && (baseProps.targetGroupName.length > 32 || !/^[0-9a-z]+[0-9a-z-]*[0-9a-z]+$/i.test(baseProps.targetGroupName))) {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts:248]: Can you extract this logic into a separate function `validateTargetGroupName`? Same for `loadBalancerName`.

**Review by corymhall — CHANGES_REQUESTED** [review]
@mina-asham just a couple of comments.

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts (hunk 2 above)
Comment (mina-asham) [reply packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts:248]: I think better, I will move it to the new `validate` method coming from the new construct API (I say "new" here, but I mean new to me 😄)

**Pull request merged**